File: Day67/Part3/buff_collapse.py
def solve():
    try:
        line = input().split()
        if not line: return
        n = int(line[0])
        
        power = list(map(int, input().split()))
        duration = list(map(int, input().split()))
        
        max_durations = {}
        for p, d in zip(power, duration):
            if p not in max_durations or d > max_durations[p]:
                max_durations[p] = d
                
        print(sum(max_durations.values()))
    except EOFError:
        pass

if __name__ == "__main__":
    solve()

File: Day67/Part3/buff_collapse.cpp
#include <iostream>
#include <vector>
#include <map>
#include <algorithm>

using namespace std;

int main() {
    int n;
    if (cin >> n) {
        vector<int> power(n);
        vector<int> duration(n);
        for (int i = 0; i < n; ++i) cin >> power[i];
        for (int i = 0; i < n; ++i) cin >> duration[i];

        map<int, int> max_durations;
        for (int i = 0; i < n; ++i) {
            if (duration[i] > max_durations[power[i]]) {
                max_durations[power[i]] = duration[i];
            }
        }

        long long total_duration = 0;
        for (auto const& [p, d] : max_durations) {
            total_duration += d;
        }

        cout << total_duration << endl;
    }
    return 0;
}
